backend/src/main/java/de/jannisaziz/backend/security/LoginService.java

```diff
@@ -0,0 +1,69 @@
+package de.jannisaziz.backend.security;
+
+import de.jannisaziz.backend.security.email.EmailService;
+import de.jannisaziz.backend.user.UserService;
+import de.jannisaziz.backend.user.User;
+import de.jannisaziz.backend.user.UserRole;
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.argon2.Argon2PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@AllArgsConstructor
+@Service
+public class LoginService {
+
+    private final EmailService emailService;
+    private final UserService userService;
+    private final Argon2PasswordEncoder argon2PasswordEncoder;
+
+    public User signIn(LoginRequest request) throws IllegalStateException, IllegalArgumentException {
+        try {
+            User user = userService.findUser(request.getUsername(), request.getEmail());
+            String encodedRequestPassword = argon2PasswordEncoder.encode(request.getPassword());
+
+            if (user.getPassword().equals(encodedRequestPassword)) return user;
+            else throw new IllegalArgumentException("Wrong Password");
+        } catch (UsernameNotFoundException | IllegalArgumentException e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    public String signUp(LoginRequest request) throws IllegalStateException {
+        try {
+            String confirmationToken = UUID.randomUUID().toString();
```

Comment:
I would think about using a jwt as confirmation token, then you wouldn't need to save it to the db